### PR TITLE
Feature/bigquery

### PR DESCRIPTION
**Pull request by rathorevaibhav**
## Summary

Documentation update for connecting to BigQuery and referring the plio-rds-to-bigquery repository

## Test Plan

- [x] Tested locally
- [ ] Tested on staging
- [ ] Tested on production

### CODE REVIEW

File: docs/DEPLOYMENT.md [hunk 1]
```diff
@@ -98,11 +100,14 @@ Follow the steps below to set up the staging environment on AWS.
...
-       3. Ignore the variables under `local web server`.
-       4. Leave the variables `CUBEJS_REDIS_URL` and `CUBEJS_REDIS_PASSWORD` as empty for now. We'll fill them up later.
+    7. Use the `.env.example` file or our [ENV guide](ENV.md) to set all the required environment variables for your container in the `Environment Variables` section.
+       1. For production, set `CUBEJS_DEV_MODE` to false.
+       2. Ignore the variables under `local web server`.
+       3. Leave the variables `CUBEJS_REDIS_URL` and `CUBEJS_REDIS_PASSWORD` as empty for now. We'll fill them up later.
+       4. For Postgres DB
+          1. Set all `CUBEJS_DB_*` variables to your AWS RDS setup. `CUBEJS_DB_TYPE` should be `postgres`.
+       5. For BigQuery
+          1. Set all `CUBEJS_DB_BQ_*` variables to your BigQuery setup. `CUBEJS_DB_TYPE` should be `bigquery`.
```
Comment (dalmia) [inline docs/DEPLOYMENT.md]: does one need to do both of these or any one of these?

File: docs/ENV.md [hunk 2]
```diff
@@ -13,9 +13,19 @@ You can create a 32-digit secret key from this website: https://generate.plus/en
 #### `CUBEJS_SCHEDULED_REFRESH_TIMER`
 If true, it enables scheduled refreshes. Can also be set to a number representing how many seconds to wait before triggering another refresh. Recommended value is true for production deployments. Defaults to false.
 
-### Database settings
+### `PLIO_ORGANIZATIONS_SCHEMAS`
+##### Postgres
+If you're using Postgres as your Cube.js database, these are the schema names within the database. Run the following query in your postgresql to retrive the list of organization schemas:
```
Comment (dalmia) [inline docs/ENV.md]:
```suggestion
If you're using Postgres as your Cube.js database, these are the schema names within the database. Run the following query in your postgresql to retrieve the list of organization schemas:
```

File: docs/ENV.md [hunk 3]
```diff
@@ -13,9 +13,19 @@ You can create a 32-digit secret key from this website: https://generate.plus/en
 #### `CUBEJS_SCHEDULED_REFRESH_TIMER`
 If true, it enables scheduled refreshes. Can also be set to a number representing how many seconds to wait before triggering another refresh. Recommended value is true for production deployments. Defaults to false.
 
-### Database settings
+### `PLIO_ORGANIZATIONS_SCHEMAS`
+##### Postgres
```
Comment (dalmia) [inline docs/ENV.md:17]:
```suggestion
#### Postgres
```

File: docs/ENV.md [hunk 4]
```diff
@@ -13,9 +13,19 @@ You can create a 32-digit secret key from this website: https://generate.plus/en
...
+If you're using Postgres as your Cube.js database, these are the schema names within the database. Run the following query in your postgresql to retrive the list of organization schemas:
+```sql
+SELECT string_agg(schema_name, ',') FROM public.organization;
+```
+
+##### BigQuery
```
Comment (dalmia) [inline docs/ENV.md:23]:
```suggestion
#### BigQuery
```

File: docs/ENV.md [hunk 5]
```diff
@@ -13,9 +13,19 @@ You can create a 32-digit secret key from this website: https://generate.plus/en
...
+If you're using BigQuery as your Cube.js database, these are the datasets within your BigQuery project. Use a comma-separated list of all the datasets.
```
Comment (dalmia) [inline docs/ENV.md]: Am I correct with my understanding here?
```suggestion
If you're using BigQuery as your Cube.js database, these are the datasets within your BigQuery project for each organization. Use a comma-separated list of all the dataset names.
```

File: docs/ENV.md [hunk 6]
```diff
@@ -13,9 +13,19 @@ You can create a 32-digit secret key from this website: https://generate.plus/en
...
+
+### Database settings (for Postgres)
 #### `CUBEJS_DB_TYPE`
-Database engine for Plio. It should always be `postgres` as added in `.env.example`. However, if you wish to add your own implementation, feel free to change it.
+Database engine for Plio. It should be `postgres` as added in `.env.example`. However, if you wish to add your own implementation, feel free to change it.
```
Comment (dalmia) [inline docs/ENV.md:28]: this is not accurate as we are now shifting to bigquery right?

File: docs/ENV.md [hunk 7]
```diff
@@ -13,9 +13,19 @@ You can create a 32-digit secret key from this website: https://generate.plus/en
...
+##### Postgres
+If you're using Postgres as your Cube.js database, these are the schema names within the database. Run the following query in your postgresql to retrive the list of organization schemas:
+```sql
+SELECT string_agg(schema_name, ',') FROM public.organization;
+```
+
+##### BigQuery
+If you're using BigQuery as your Cube.js database, these are the datasets within your BigQuery project. Use a comma-separated list of all the datasets.
+
+### Database settings (for Postgres)
```
Comment (dalmia) [inline docs/ENV.md]: from what I understand, we are not using this right? Would be good to say that clearly here.

File: docs/ENV.md [hunk 8]
```diff
@@ -34,6 +44,27 @@ The database user.
 #### `CUBEJS_DB_PASS`
 The password for the database user.
 
+### Database settings (for BigQuery)
+You may wish to use BigQuery for your staging or production setup. However, if you want to connect from your local setup, you can do that as well. Configure the following env variables to connect your Cube.js instance to your BigQuery instance. These variables are not defined in `.env.example`.
```
Comment (dalmia) [inline docs/ENV.md]: these variables should be defined in `.env.example` no?

File: docs/ENV.md [hunk 9]
```diff
@@ -34,6 +44,27 @@ The database user.
...
+#### `CUBEJS_DB_BQ_CREDENTIALS`
+This is a base64 encoded value of your Google Cloud Platform's service account. You can learn more about acquiring service account credentials [here](https://cloud.google.com/docs/authentication/getting-started) and [here](https://console.cloud.google.com/projectselector2/iam-admin/serviceaccounts?supportedpurview=project). The service account must have BigQuery admin permissions.
+
+Once you have downloaded the JSON file, run the following commands and use the output for this environment variable:
+```sh
+cat /path/to/gcp-service-account-filename.json | base64
+```
+
+#### `CUBEJS_DB_BQ_LOCATION`
+The region of your bigquery setup in Google Cloud Console. You can check this from the dataset info section mentioned above. You can find all supported regions [here](https://cloud.google.com/bigquery/docs/locations#regional-locations).
```
Comment (dalmia) [inline docs/ENV.md]: I don't see any dataset info section mentioned above - am I missing something?

File: docs/ENV.md (hunk 7 above)
Comment (rathorevaibhav) [reply docs/ENV.md]: I have updated it. Let me know if it makes sense.

File: docs/ENV.md (hunk 8 above)
Comment (rathorevaibhav) [reply docs/ENV.md]: Added.

File: docs/DEPLOYMENT.md (hunk 1 above)
Comment (rathorevaibhav) [reply docs/DEPLOYMENT.md]: Just one. Updated.

File: docs/ENV.md (hunk 9 above)
Comment (rathorevaibhav) [reply docs/ENV.md]: oh i think I had it added, then removed. Updating this statement.

File: docs/DEPLOYMENT.md [hunk 10]
```diff
@@ -104,12 +104,14 @@ Follow the steps below to set up the staging environment on AWS.
        1. For production, set `CUBEJS_DEV_MODE` to false.
        2. Ignore the variables under `local web server`.
        3. Leave the variables `CUBEJS_REDIS_URL` and `CUBEJS_REDIS_PASSWORD` as empty for now. We'll fill them up later.
-       4. For Postgres DB
-          1. Set all `CUBEJS_DB_*` variables to your AWS RDS setup. `CUBEJS_DB_TYPE` should be `postgres`.
-       5. For BigQuery
-          1. Set all `CUBEJS_DB_BQ_*` variables to your BigQuery setup. `CUBEJS_DB_TYPE` should be `bigquery`.
-    8.  Save the container definition and the task definition.
```
Comment (dalmia) [inline docs/DEPLOYMENT.md:106]: guessing that this line got deleted by mistake.

File: docs/ENV.md [hunk 11]
```diff
@@ -14,38 +14,17 @@ You can create a 32-digit secret key from this website: https://generate.plus/en
 If true, it enables scheduled refreshes. Can also be set to a number representing how many seconds to wait before triggering another refresh. Recommended value is true for production deployments. Defaults to false.
 
 ### `PLIO_ORGANIZATIONS_SCHEMAS`
```
Comment (dalmia) [inline docs/ENV.md]: I had made my earlier suggestions for heading based on this - saw the file now and realized that the mistake is here.
```suggestion
#### `PLIO_ORGANIZATIONS_SCHEMAS`
```

File: docs/ENV.md [hunk 12]
```diff
@@ -14,38 +14,17 @@ You can create a 32-digit secret key from this website: https://generate.plus/en
...
-##### Postgres
-If you're using Postgres as your Cube.js database, these are the schema names within the database. Run the following query in your postgresql to retrive the list of organization schemas:
+#### Postgres
```
Comment (dalmia) [inline docs/ENV.md]:
```suggestion
##### Postgres
```

File: docs/ENV.md [hunk 13]
```diff
@@ -14,38 +14,17 @@ You can create a 32-digit secret key from this website: https://generate.plus/en
...
-
-#### `CUBEJS_DB_PORT`
-The port for the database.
-
-#### `CUBEJS_DB_USER`
-The database user.
-
-#### `CUBEJS_DB_PASS`
-The password for the database user.
+#### BigQuery
```
Comment (dalmia) [inline docs/ENV.md]:
```suggestion
##### BigQuery
```

**Pull request merged**